tests/unit/state-getters.spec.ts

```diff
@@ -180,3 +143,157 @@ describe('State Getters', () => {
+    store.state.stateModel.nameRequest.legalName = 'MyLegalName'
+    expect(vm.hasBusinessNameChanged).toBe(false)
+
+    // verify that business type changes are detected
+    store.state.stateModel.tombstone.entityType = 'BEN2'
+    expect(vm.hasBusinessTypeChanged).toBe(true)
+    store.state.stateModel.tombstone.entityType = 'BEN'
+    expect(vm.hasBusinessTypeChanged).toBe(false)
+
+    // verify that name translation changes are detected
+    store.state.stateModel.nameTranslations = [{ action: 'ACTION' }]
+    expect(vm.hasNameTranslationChanged).toBe(true)
+    store.state.stateModel.nameTranslations = []
+    expect(vm.hasNameTranslationChanged).toBe(false)
+
+    // verify that registered mailing address changes are detected
+    store.state.stateModel.officeAddresses = { registeredOffice: { mailingAddress: { postalCode: 'V8V 8V8' } } }
+    expect(vm.haveOfficeAddressesChanged).toBe(true)
+    store.state.stateModel.officeAddresses = null
+    expect(vm.haveOfficeAddressesChanged).toBe(false)
+
+    // verify that registered delivery address changes are detected
+    store.state.stateModel.officeAddresses = { registeredOffice: { deliveryAddress: { postalCode: 'V8V 8V8' } } }
+    expect(vm.haveOfficeAddressesChanged).toBe(true)
+    store.state.stateModel.officeAddresses = null
+    expect(vm.haveOfficeAddressesChanged).toBe(false)
+
+    // verify that records mailing address changes are detected
+    store.state.stateModel.officeAddresses = { recordsOffice: { mailingAddress: { postalCode: 'V8V 8V8' } } }
+    expect(vm.haveOfficeAddressesChanged).toBe(true)
+    store.state.stateModel.officeAddresses = null
+    expect(vm.haveOfficeAddressesChanged).toBe(false)
+
+    // verify that records delivery address changes are detected
+    store.state.stateModel.officeAddresses = { recordsOffice: { deliveryAddress: { postalCode: 'V8V 8V8' } } }
+    expect(vm.haveOfficeAddressesChanged).toBe(true)
+    store.state.stateModel.officeAddresses = null
+    expect(vm.haveOfficeAddressesChanged).toBe(false)
+
+    // verify that people and roles changes are detected
+    store.state.stateModel.peopleAndRoles.orgPeople = [{}]
+    expect(vm.havePeopleAndRolesChanged).toBe(true)
+    store.state.stateModel.peopleAndRoles.orgPeople = []
+    expect(vm.havePeopleAndRolesChanged).toBe(false)
+
+    // verify that share structure changes are detected
+    store.state.stateModel.shareStructureStep.shareClasses = [{}]
+    expect(vm.hasShareStructureChanged).toBe(true)
+    store.state.stateModel.shareStructureStep.shareClasses = []
+    expect(vm.hasShareStructureChanged).toBe(false)
+
+    // verify that incorporation agreement changes are detected
+    store.state.stateModel.incorporationAgreementStep.agreementType = 'custom'
+    expect(vm.hasIncorporationAgreementChanged).toBe(true)
+    store.state.stateModel.incorporationAgreementStep.agreementType = 'sample'
+    expect(vm.hasIncorporationAgreementChanged).toBe(false)
```

Comment:
Yes, this is repetitive, but I think a struct to drive a loop would be just as ugly, so this is at least understandable :)